simplex.md

```diff
@@ -2,230 +2,84 @@
-  - store history of all conversations encrypted using user client app key with
-    passphrase (or some other device specific encryption method).
-
-## System design
-
-The chat system design is based on 2 protocols, each with the generic part,
-describing protocol flow and logic, and implementation part, describing protocol
-transports, data structures and algorithms.
-
-1. [simplex messaging protocol][6] - a low level messaging protocol that defines
-   establishing and using a simplex connection between chat participants on a
-   single server. While this protocol is designed to support graph-chat client
-   protocol (below), it can be used for other messaging scenarios, not limited
-   to chats.
-2. [graph-chat protocol][8] - a high level generic chat protocol for client
-   applications (graph vertices) that communicate via simplex connections
-   created using simplex messaging protocol. This protocol defines connection
-   and message types and semantics for:
-   - various chat elements (user profiles, direct chats, chat groups,
-     broadcasts, etc.).
-   - other communication scenarios - e.g. introduction, off-the-record chat,
-     etc.
-   - using multiple servers to ensure message delivery.
-   - sharing user profiles, contacts and chats across multiple client devices.
-   - changing cryptographic keys and servers used to send and receive messages
-     using simplex messaging server protocol.
-   - sending and receiving out-of-band messages between client applications
-     using "visual code".
-3. graph-chat client application protocol (TODO) - a high level specific chat
-   protocol for client applications. This protocol specifies:
-   - data structures for sending and receiving messages of all types.
-   - process to send and receive out-of-band messages between client
-     applications.
-   - other requirements for graph-chat client applications.
-   - defines a specific "visual code" format to send an out-of-band message.
+  - reduce the risk of attacker posing as user's connection;
+  - avoid exposing all user connections to the servers.
+- Unique public key is used to identify each connection participant to each server.
+- Public keys used between connections are regularly rotated to prevent decryption of the full message history ([forward secrecy][4]) in case when some servers or middlemen preserve message history and the current key is compromised.
+- Users can repeat key exchange using QR code and alternative channel at any point to increase communication security and trust.
+- No single server in the system has visibility of all connections or messages of any user, as user profiles are identified by multiple rotating public keys, using separate key for each profile connection.
+- User profile (meta-data of the user including non-unique name / handle and optional additional data, e.g. avatar and status) is stored in the client apps and is shared only with accepted user profile connections.
```

Comment:
merge this with
`Only client applications store user profiles, contacts of other user profiles, messages; servers do NOT have access to any of this information and (unless compromised) do NOT store encrypted messages or any logs.`
?

Reply:
removed, it's almost the same